module/userProjects/helper.js

```diff
@@ -1749,569 +1763,560 @@ module.exports = class UserProjectsHelper {
+                    success: true,
+                    message: CONSTANTS.apiResponses.UPDATED_DOCUMENT_SUCCESSFULLY,
+                    data: result
+                });
 
-            let query = {
-                userId : userId,
-                isDeleted : false,
-                programId : { $exists : true }
+            } catch (error) {
+                return resolve({
+                    status:
+                        error.status ?
+                            error.status : HTTP_STATUS_CODE['internal_server_error'].status,
+                    success: false,
+                    message: error.message,
+                    data: {}
+                });
             }
+        });
+    }
 
-            let searchQuery = [];
-            let filterQuery = [];
+    /**
+* Get list of user projects with the targetted ones.
+* @method
+* @name getProject 
+* @param {String} userId - Logged in user id.
+* @param {String} userToken - Logged in user token.
+* @returns {Object}
+*/
 
-            if (search !== "") {
-                searchQuery = [
-                    { "title" : new RegExp(search, 'i') },
-                    { "description" : new RegExp(search, 'i') }
-                ];
-            }
+    static getProject(bodyData, userId, userToken, pageSize, pageNo, search, filter) {
+        return new Promise(async (resolve, reject) => {
+            try {
+
+                let query = {
+                    userId: userId,
+                    isDeleted: false
+                }
 
-            if (filter && filter !== "") {
-                if(filter == CONSTANTS.common.ASSIGN_TO_ME){
+                let searchQuery = [];
 
-                    filterQuery = [
-                        { isAPrivateProgram : false }
+                if (search !== "") {
+                    searchQuery = [
+                        { "title": new RegExp(search, 'i') },
+                        { "description": new RegExp(search, 'i') }
                     ];
+                }
 
-                }else if(filter == CONSTANTS.common.CREATED_BY_ME){
+                bodyData["filter"] = {};
 
-                    filterQuery = [
-                        { isAPrivateProgram: { $ne: false }} 
-                    ];
+                if (filter && filter !== "") {
+                    if (filter === CONSTANTS.common.CREATED_BY_ME) {
+                        query["isAPrivateProgram"] = bodyData["filter"]["isAPrivateProgram"] = {
+                            "$ne": false
+                        };
+                    } else if (filter == CONSTANTS.common.ASSIGN_TO_ME) {
+                        query["isAPrivateProgram"] = bodyData["filter"]["isAPrivateProgram"] = false;
+                    }
                 }
 
-                query = {...query, ...filterQuery[0]};
-                
-            }
+                let projects = await this.projects(
+                    query,
+                    CONSTANTS.common.DEFAULT_PAGE_SIZE,
```

Comment:
@aman-tunerlabs what is someone whats an another value for page size and not the default one?

Reply:
@aks30  This logic works for pagination having search field and page number . Pagination in this api should work not just for user assigned project but should also work for the one that is targeted. So Fetching all the assigned project and combining the targeted ones and finally doing pagination on the merged data.

Reply:
@aks30  Current apis works with pagination so no issue on that.